commands/lib/install.go

```diff
@@ -73,7 +75,24 @@ func LibraryInstall(ctx context.Context, req *rpc.LibraryInstallRequest, downloa
 		if err != nil {
 			return err
 		}
+		libReleasesToInstall = append(libReleasesToInstall, libRelease)
+	}
 
+	// Check if any of the libraries to install is already installed
+	if req.GetNoOverwrite() {
+		for _, libRelease := range libReleasesToInstall {
+			_, libReplaced, err := lm.InstallPrerequisiteCheck(libRelease)
+			if err != nil {
+				return err
```

Comment:
`github.com/arduino/arduino-cli/arduino/libraries/librariesmanager.ErrAlreadyInstalled` is returned whenever a dependency is already installed:

https://github.com/arduino/arduino-cli/blob/35834c50fd8e81f52058a2fbce03e6dbd9953c81/arduino/libraries/librariesmanager/install.go#L59-L61

this causes the installation to fail even if it would not result in an upgrade:

```text
$ arduino-cli version
arduino-cli.exe  Version: git-snapshot Commit: 35834c50 Date: 2022-07-08T04:58:38Z

$ arduino-cli lib install SD
Downloading SD@1.2.4...
SD@1.2.4 already downloaded
Installing SD@1.2.4...
Installed SD@1.2.4

$ arduino-cli lib install --no-overwrite Arduino_BuiltIn
Error installing Arduino_BuiltIn: library already installed

$ arduino-cli lib list
Name Installed     Available         Location              Description
SD   1.2.4         -                 LIBRARY_LOCATION_USER -
```

My expectation is that installation of a dependency would be skipped if the currently installed version of that dependency meets the version constraint specified by the dependent library, but that the installation of the dependent library and any other of its dependencies should proceed.

Reply:
Uhm... the implementation it's a bit messy, let me see if I can clean up a bit

Reply:
Ok, I've updated the PR, now the "already-installed" state is checked prior to installation, this will improve the experience when the user tries to update an already updated library, before it was:

```
$ arduino-cli lib install SD
Downloading SD@1.2.4...
 SD@1.2.4 già scaricato
Installazione SD@1.2.4...
Già installato SD@1.2.4
```

now is:

```
$ arduino-cli lib install SD
Già installato SD@1.2.4
```

so it prints directly the "already installed" message without even trying to download.
Hopefully, I did not break anything else... :-)